### DISCUSSION TITLE
Multi-cursor support is missing

### DISCUSSION THREAD

**Discussion by sedrubal** [community/community#7942] [Copilot Conversations]
When using multi-cursor mode in vscode, suggestions are only available for one cursor.

**Comment by freeall** [comment]
Here is an example where the suggestion only show up where the first cursor is, and not the next.

![image](https://user-images.githubusercontent.com/376654/143439801-32c8a49b-2910-4baf-9a50-7f67079e5f54.png)

**Reply by ggoonnzzaallo** [reply]
@freeall idk how this still has not been implemented by Microsoft 2 years+ later but I built an extension that supports multi-cursor. You can try it [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot) if interested. Example pic below.


![Screenshot 2024-02-28 at 4 33 45 PM](https://github.com/community/community/assets/121972034/21385d75-87a6-4a6e-bdeb-2ef2829f9187)

**Reply by tendiesfortwo** [reply]
@ggoonnzzaallo looks cool, will check it out!

**Comment by martin-marko** [comment]
This sounds like a lot of work to implement (properly) but would be awesome

**Comment by zolastro** [comment]
Personally, I think it would be enough just to copy the recommended solution of the first cursor. Most of the time I just want to paste that suggestion to all of my cursors. Furthermore, if each of the cursors gave a different suggestion, it would be a pain to check all of them before accepting them.

**Reply by Arturlang** [reply]
Exactly, I think just copying the suggestion is just fine

**Reply by sedrubal** [reply]
But if you have repetitive code, which for example should add +1 in one line and +2 in the other one, copying code would not help much. Instead it should suggest the same code structure with adapted values for each cursor.

**Reply by lachieh** [reply]
This is pretty much what I want it for also. Just had a situation where I highlighted a function name in a few places in the file and the suggested new name was perfect, but it only placed it at the first cursor.

**Reply by matthew-mcallister** [reply]
It would be useful to have different suggestions for each cursor in cases like passing a bunch of parameters to a function or dictionary. E.g.
```
my_func(
    foo=foo,
    bar=bar,
    ...
)
```

**Reply by steviebuilds** [reply]
Doing a straight copy-paste wouldn't really be aligned with what the project is about. Surely it would have the capability to increment the items if there was a pattern or just offer the same item if there is nothing "incremental" about the context.

**Comment by torshepherd** [comment]
Bump. I would expect Copilot to simply behave as if I had highlighted my cursor at the first location, selected that suggestion, and then placed my cursor at the subsequent location.

Edge cases I could see would be if the multi-cursors are close together. Can anyone else think of anything else?

**Comment by hrdwdmrbl** [comment]
I do personally expect each cursor to give a different suggestion. Often I'm doing something similar but slightly different at each. For example, creating a getter method for multiple different properties.

But, if there are too many edge cases I'm find with just a single suggestion being filled at every cursor.

**Comment by hamishjam** [comment]
+1 for this.  I wouldn't have thought multi-cursor support would be super difficult to implement. The challenge will be how to accept the suggestions as I would assume that each cursor would prompt its own AI get.  You wouldn't want to mass accept the suggestions as you potentially wouldn't see them all without scrolling. Perhaps the option to select en masse or one-by-one?

**Comment by ggoonnzzaallo** [comment]
After 2 years+, got tired of waiting for Microsoft to fix this so I just built my own extension to handle multi-cursor autocompletes. If anyone is interested you can try it [here](https://marketplace.visualstudio.com/items?itemName=doublebot.doublebot)

![Screenshot 2024-02-28 at 4 33 45 PM](https://github.com/community/community/assets/121972034/5cbdd293-ca15-46a2-a4f3-86189902964d)
